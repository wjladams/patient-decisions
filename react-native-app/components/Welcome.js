 import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

export default class Welcome extends React.Component {

    render() {
        const {navigate} = this.props.navigation;

        return (
            <View style={styles.container}>
                <Text style={styles.welcomeText}>WeDecide</Text>
                <Text style={styles.startButton} onPress = {() => navigate('PatientForm')}>start here</Text>
                <Text></Text>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'space-between',
        alignItems: 'center'
    },
    welcomeText: {
        fontSize: 50,
        paddingTop: 20,
        justifyContent: 'flex-start',
        alignItems: 'center'
    },
    startButton: {
        fontSize: 30,
        margin: 10,
        padding: 10,
        borderWidth: 2,
        borderRadius: 10,
        justifyContent: 'center',
        alignItems: 'center'
    }
});