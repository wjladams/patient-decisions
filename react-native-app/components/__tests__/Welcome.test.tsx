import 'react-native';
import React from "react";
import renderer from "react-test-renderer"
import { Welcome } from "../Welcome";
import { NavigationScreenProp } from "react-navigation";

test("renders correctly with defaults", () => {
  const mockNavigation: NavigationScreenProp<any, any> = { navigate: jest.fn } as any;
  const button = renderer.create(<Welcome navigation={mockNavigation} />).toJSON()
  expect(button).toMatchSnapshot()
})