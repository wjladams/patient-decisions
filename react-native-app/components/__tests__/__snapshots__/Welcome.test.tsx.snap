// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-between",
    }
  }
>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 50,
        "justifyContent": "flex-start",
        "paddingTop": 20,
      }
    }
  >
    WeDecide
  </Text>
  <Text
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "borderWidth": 2,
        "fontSize": 30,
        "justifyContent": "center",
        "margin": 10,
        "padding": 10,
      }
    }
  >
    start here
  </Text>
  <Text />
</View>
`;
